### ISSUE TITLE
OneToOne eager not loaded

### ISSUE THREAD

**Issue by kristijanhusak** [mikro-orm/mikro-orm#2777]
**Describe the bug**
I noticed while updating to v5 that OneToOne relationships with `eager: true` does not load some of them. I have a user entity that has one to one to several other entities, and only 2 of them gets loaded.

**To Reproduce**
1. Clone https://github.com/kristijanhusak/mikroorm-migration-issue
2. Start with `docker-compose up`
3. open `http://localhost:3000/custom`
4. Check the query in console, it's missing a join for `Comment` entity.

Link to entity that is being queried by this endpoint: https://github.com/kristijanhusak/mikroorm-migration-issue/blob/master/src/entities/customer.entity.ts

**Expected behavior**
Expected to get all of the eager relationships joined

**Additional context**
This works fine in v4.5, and it also works fine if I add `LoadStrategy.JOINED` to all of the `OnetoOne` definitions in the customer entity.

**Versions**

| Dependency | Version |
| - | - |
| node | 16.13.0 |
| typescript | 4.55 |
| mikro-orm | 5.0.1 |
| your-driver | postgres |

**Comment by kristijanhusak** [comment]
This is the copy of the environment where I got this issue. I don't know if it would happen with sqlite, or any other setup.
Indexes are left from my previous things I tried to reproduce, I'm not sure why that's an issue. I missed to write a step where you need to run the migrations, but migrations have it all covered. 

Docker is much simpler for setting up reproduction steps because everything is contained. I'm not even sure if I have sqlite installed on my machine, and expecting to have postgres installed for someone else would be even worse.

As an open source contributor I understand that you need reproduction steps, and I'm providing those, even if it's not perfect.
I don't want to spend bunch of time setting up ideal reproduction steps. Additionally, there is a workaround I mentioned, so this is not a very critical issue, so running the reproduction steps, even if it takes some time, can wait.

> no repositories (sometimes the problem is actually with using repository, by doing this you find it out yourself)

Aren't the repositories part of the Mikro ORM core? Why wouldn't those be considered? Repositories are the only thing I use in my application to communicate with DB. Using ORM instance or entity manager directly seems like a bad practice.

**Comment by B4nan** [comment]
I am sorry but those are all just excuses. FYI I just spend more than hour setting up a reproduction based on your project. Just to see if it is an actual bug or not. If I had a reproduction already, I could spend that hour fixing it. What do you want me to spend my time on, doing the job you can (and should) help with, or fixing the bug?

> I don't know if it would happen with sqlite, or any other setup.

That is exactly why _you_ should try it - to help me debug this problem - to help me debug your problem. 

>  I'm not sure why that's an issue

Because it adds complexity to the reproduction. 

> I missed to write a step where you need to run the migrations, but migrations have it all covered.

That is another thing that should not be part of reproduction. It again just adds complexity instead of narrowing the problem. Ideally there should be just one line setting up the schema  (`orm.getSchemaGenerator().refreshDatabase()`).

>  I'm not even sure if I have sqlite installed on my machine,

Yes you have, like everybody with windows, macos or linux. 

> I don't want to spend bunch of time setting up ideal reproduction steps.

But I can spend that bunch of time, right? 

> Aren't the repositories part of the Mikro ORM core? Why wouldn't those be considered?

I said you should try to see if the problem is in using repositories or not. If its not there, don't use them in reproduction, as they again just add complexity.

**Comment by kristijanhusak** [comment]
> FYI I just spend more than hour setting up a reproduction based on your project

Which part took the most? The steps I missed, or some other? Docker should spin up in a minute or so. 
If you have stuck on any of the steps, letting me know would be perfectly fine. I didn't forget those steps on purpose.

I didn't mean to spend bunch of your time setting up the reproduction steps. I tried to have as less time spent as possible for both you and me.

**Comment by kristijanhusak** [comment]
I'm getting only this query logged:

```sql
select
  "c0"."id",
  "c0"."first_name",
  "c0"."last_name",
  "c0"."created_at",
  "c0"."updated_at",
  "p1"."id" as "p1__id",
  "p1"."title" as "p1__title",
  "p1"."description" as "p1__description",
  "p1"."image_id" as "p1__image_id",
  "p1"."customer_id" as "p1__customer_id",
  "p1"."created_at" as "p1__created_at",
  "p1"."updated_at" as "p1__updated_at",
  "i2"."id" as "i2__id",
  "i2"."image_id" as "i2__image_id",
  "i2"."customer_id" as "i2__customer_id",
  "p1"."id" as "product_id",
  "p3"."id" as "post_id"
from
  "customers" as "c0"
  left join "products" as "p1" on "c0"."id" = "p1"."customer_id"
  left join "images" as "i2" on "p1"."image_id" = "i2"."id"
  left join "posts" as "p3" on "c0"."id" = "p3"."customer_id"
```

It does left join for products and posts, which also doesn't have a `joined` strategy attached, so I assumed same thing should happen for comment.

If comment is fetched separately because it's not a `joined` strategy, I would expect to get that query also logged into console besides the above one, but there isn't any other queries.

I'll give this fix a test with a 5.0.3 release in my project where I can test it more thoroughly.
Thanks.
